syncd/syncd_notifications.cpp

```diff
@@ -461,8 +461,68 @@ void processNotification(
 // that some notiffication arrived
 
 std::condition_variable cv;
-std::mutex queue_mutex;
-std::queue<swss::KeyOpFieldsValuesTuple> ntf_queue;
+
+class ntf_queue_t
+{
+public:
+    bool enqueue(swss::KeyOpFieldsValuesTuple msg);
+    bool tryDequeue (swss::KeyOpFieldsValuesTuple& msg);
+    size_t  queueStats()
+    {
+        return ntf_queue.size();
```

Comment:
this needs to be protected by mutex also, in most cases this should be fine, but in some corner case it may hit some issue ? not sure about internal implementation of size() but is it's a single field lookup then this is fine without locking